### PR TITLE
Hometasks/hometask02

### PR DESCRIPTION
**Pull request by Frtn0717**
Я так понял, что автоматически не может смержиться потому что я readme файл исправил отдельно в main и отдельно в hometasks? Это же не критично в данном случае? Но лучше так не делать, верно?

### CODE REVIEW

File: hometask02/coercion.js [hunk 1]
```diff
@@ -0,0 +1,296 @@
+const str = document.getElementById('string');
+const bool = document.getElementById('boolean');
+const num = document.getElementById('number');
+const obj = document.getElementById('objects');
+
+str.addEventListener('click', function openCloseString(event) {
```
Comment (ArtsiomFinenka) [inline hometask02/coercion.js:6]: You use some ES6 features ( "const"), but don't use the arrow function as a callback. Let's use it here and below. And you shouldn't name the callback function here.

File: hometask02/coercion.js [hunk 2]
```diff
@@ -0,0 +1,296 @@
...
+        item.setAttribute('disabled', 'disabled');
+        item.classList.add('show');
+        target.value = 'Hide result';
+      }
+    }
+
+    for(item of convField) {
+
+      if(item.querySelector('.string_number_result')) {
+        item.querySelector('.string_number_result').value = "'" + String(2021) + "'";
```
Comment (ArtsiomFinenka) [inline hometask02/coercion.js:25]: The same, let's use ES6 Template strings here and below.

File: hometask02/coercion.js [hunk 3]
```diff
@@ -0,0 +1,296 @@
...
+        item.classList.add('show');
+        target.value = 'Hide result';
+      }
+    }
+
+    for(item of convField) {
+
+      if(item.querySelector('.objects_boolean_result')) {
+        item.querySelector('.objects_boolean_result').value = 'true + false  ==> 1 + 0  ==> 1';
+      }
```
Comment (ArtsiomFinenka) [inline hometask02/coercion.js:221]: IFor each case in the section "Type coercion for object":
- add the current value into HTML markup as "explanation". For example: 
image
- to generate value use the real code (like in the previous sections). For example:
item.querySelector('.objects_boolean_result').value = `${true + false}`;
image

File: hometask02/index.html [hunk 4]
```diff
@@ -0,0 +1,538 @@
...
+        </code>
+      </article>
+       <br />
+
+      <article class="conv_field" id="string_number">
+        <code>
+          String(2021) = 
+        </code>
+        <input type="text" class="string_number_result hidden" id="str_num_res" />
+        <label for="str_num_res"></label>
```
Comment (ArtsiomFinenka) [inline hometask02/index.html:32]: I see all labels are empty. Why do you use it? Please delete it in the whole file if label is empty

File: hometask02/coercion.js [hunk 5]
```diff
@@ -0,0 +1,296 @@
+const str = document.getElementById('string');
```
Comment (ArtsiomFinenka) [inline hometask02/coercion.js:1]: This comment is not related to this string.
I found a bug. Steps to reproduce:
- Click "Show result"
- Click on the "code" or previous "input" tag
image
- The input will be hidden, and the button will still have the value "Hide Result":
image

File: hometask02/coercion.js [hunk 6]
```diff
@@ -0,0 +1,296 @@
...
+  const target = event.target;
+  const resultHidden = document.querySelectorAll('.hidden');
+  const resultShow = document.querySelectorAll('.show');
+  const convField = document.querySelectorAll('.conv_field');
+
+  if(target.value == 'Show result') {
+    for(item of resultHidden) {
+      if(target.parentNode == item.parentNode) {
+        item.classList.remove('hidden');
+        item.setAttribute('disabled', 'disabled');
```
Comment (ArtsiomFinenka) [inline hometask02/coercion.js:16]: Why do you disable this in the JS code. It is better to disable in the HTML markup

**Review by ArtsiomFinenka — CHANGES_REQUESTED** [review]

File: hometask02/index.html (hunk 4 above)
Comment (Frtn0717) [reply hometask02/index.html:32]: My bad. Just I thought if I use input then I need to use label for this input.

**Review by ArtsiomFinenka — APPROVED** [review]
You can merge it and resolve the conflict

**Pull request merged**